src/Symfony/Component/HttpFoundation/Cookie.php

```diff
@@ -141,7 +143,15 @@ public function __construct(string $name, string $value = null, $expire = 0, ?st
      */
     public function __toString()
     {
-        $str = ($this->isRaw() ? $this->getName() : urlencode($this->getName())).'=';
+        if ($this->isRaw()) {
+            $str = $this->getName();
+        } else {
+            $reservedChar = str_split(self::RESERVED_CHARS);
+            $reservedCharsEncoded = array_map('rawurlencode', $reservedChar);
```

Comment:
let's make these a local `static` property to save dong the computation too often?

Reply:
I've fixed. Is it ok now?

Thank you for your review! Your comments are very helpful.